File: pompo-backend/app/schemas/health.py
"""Health check response schemas."""

from typing import Any

from pydantic import BaseModel, Field


class ComponentHealth(BaseModel):
    """Health status of an individual subsystem."""

    status: str
    healthy: bool
    details: dict[str, Any] | None = None


class HealthResponse(BaseModel):
    """Comprehensive health check response."""

    status: str
    version: str
    environment: str
    components: dict[str, ComponentHealth]
    response_time_ms: float = Field(description="Total health check duration in milliseconds")


class ReadinessResponse(BaseModel):
    """Readiness probe response."""

    ready: bool
    checks: dict[str, bool]


class LivenessResponse(BaseModel):
    """Liveness probe response."""

    alive: bool

File: pompo-frontend/src/lib/types/health.ts
/** Mirrors app/schemas/health.py */

export interface ComponentHealth {
  status: string;
  healthy: boolean;
  details?: Record<string, unknown> | null;
}

export interface HealthResponse {
  status: string;
  version: string;
  environment: string;
  components: Record<string, ComponentHealth>;
  response_time_ms: number;
}

export interface ReadinessResponse {
  ready: boolean;
  checks: Record<string, boolean>;
}

export interface LivenessResponse {
  alive: boolean;
}
